src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -518,7 +520,7 @@ private function createTypeAlternatives(TypedReference $reference)
         } elseif ($reference->getRequiringClass() && !$reference->canBeAutoregistered() && !$this->strictMode) {
             return ' It cannot be auto-registered because it is from a different root namespace.';
         } else {
-            return;
+            return null;
         }
 
         return sprintf(' You should maybe alias this %s to %s.', class_exists($type, false) ? 'class' : 'interface', $message);
```

Comment:
This actually looks broken: The return statement here is unreachable.

Reply:
yes it is: see first if or second elseif

Reply:
Okay, I didn't see that the first two `if`s are not returning. 🙈